component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java

```diff
@@ -1688,4 +1764,222 @@ public static RelationshipManager getRelationshipManager() {
+                                                                                          temporalCheck,
+                                                                                          excludeGroupNodesWithoutPageChildNodes,
+                                                                                          locale))
+                                                             .filter(Objects::nonNull)
+                                                             .toList();
+    return sortByDisplayOrder ? siteEntities
+                              : siteEntities.stream().sorted(Comparator.comparing(SiteEntity::getDisplayName, String.CASE_INSENSITIVE_ORDER)).toList();
+  }
+
+  public static SiteEntity buildSiteEntity(PortalConfig site,
+                                           HttpServletRequest request,
+                                           boolean expandNavigations,
+                                           List<String> visibilityNames,
+                                           boolean excludeEmptyNavigationSites,
+                                           boolean temporalCheck,
+                                           boolean excludeGroupNodesWithoutPageChildNodes,
+                                           Locale locale) {
+    if (site == null) {
+      return null;
+    }
+    SiteType siteType = SiteType.valueOf(site.getType().toUpperCase());
+    String displayName = site.getLabel();
+    org.exoplatform.services.security.Identity userIdentity = ConversationState.getCurrent().getIdentity();
+    UserNode rootNode = null;
+    String currentUser = userIdentity.getUserId();
+    try {
+      HttpUserPortalContext userPortalContext = new HttpUserPortalContext(request);
+      UserPortalConfig userPortalConfig =
+                                        getUserPortalConfigService().getUserPortalConfig(siteType != SiteType.PORTAL ? getUserPortalConfigService().getDefaultPortal()
+                                                                                                                     : site.getName(),
+                                                                                         currentUser,
+                                                                                         userPortalContext);
+
+      UserPortal userPortal = userPortalConfig.getUserPortal();
+      UserNavigation navigation = userPortal.getNavigation(new SiteKey(siteType.getName(), site.getName()));
+      if (navigation != null) {
+        UserNodeFilterConfig.Builder builder = UserNodeFilterConfig.builder();
+        builder.withReadWriteCheck().withVisibility(convertVisibilities(visibilityNames));
+        if (temporalCheck) {
+          builder.withTemporalCheck();
+        }
+        rootNode = userPortal.getNode(navigation, Scope.ALL, builder.build(), null);
+      }
+    } catch (Exception e) {
+      LOG.error("Error while getting site {} navigations for user {}", site.getName(), currentUser, e);
```

Comment:
Why not throwing the error, knowing that it's blocker to build a consistent information to return to the user

Reply:
IMO it is not blocker, because even if when we catch the exception and just log it without throwing, the site Navigations will be null and not returned. In that case, the site will be returned just without the navigations which is not an optional information.

Reply:
Yep, but to diagnose and be aware that the REST API had errors, we will have to ask support or ITOP teams to check on it, while the REST API should be sel explanatory (see REST Best practices) ...

Reply:
ok we throw an ObjectNotFoundException in that case to the Rest ?

Reply:
Nope, I think it's an unkown error, thus an HTTP 500 with `e.getMessage()` as entity + log it as WARN and not error since it's not a general Server status Error.